src/ua_types.c

```diff
@@ -115,6 +117,26 @@ UA_cleanupDataTypeWithCustom(const UA_DataTypeArray *customTypes) {
 /* Builtin Types */
 /*****************/
 
+UA_String
+UA_String_fromFormat (char *format, ...)
+{
+    UA_String s;
+    s.data = NULL;
+    va_list args;
+    va_list args_copy;
+
+    va_start (args, format);
```

Comment:
I suggest you first encode into a 256 byte stack-allocated buffer first.
When that succeeds you can memcpy and don’t have to encode a second time.